### PR TITLE
Fix LSP adapter errors on tab close

### LINKED ISSUE #13917: LSP adapter throws console errors when closing tabs sometimes
## Description
If you open several connected sessions that have LSP functionality and close them, they may throw console errors because their virtual documents are unavailable and accessed anyway.

https://user-images.githubusercontent.com/159529/216573641-ed90a7d4-0ebf-4243-ab41-ca2a16942a56.mov

cc: @trungleduc

### PR DESCRIPTION
**Pull request by afshin**
This PR sets the type of the adapter accessor for virtual documents to be `VirtualDocument | null` and simplifies some of the initialization and dispose logic.

## References
Fixes #13917

## Code changes
- Updates the LSP adapter's `virtualDocument` signature to include `null`.
- Changes `ILSPDocumentConnectionManager.unregisterDocument` to accept a `uri` string instead of a full `VirtualDocument`

## User-facing changes
**Before this PR**

https://user-images.githubusercontent.com/159529/216574709-e9b0fb36-aaa8-4859-b695-6136c2ac9740.mov

**After this PR**

https://user-images.githubusercontent.com/159529/216574853-177edc9f-cd79-41c0-8be8-41df5078676e.mov

## Backwards-incompatible changes
- `WidgetLSPAdapter`
  - **Before:** `get virtualDocument(): VirtualDocument`
  - **After:** `get virtualDocument(): VirtualDocument | null`
- `ILSPDocumentConnectionManager`
  - **Before:** `unregisterDocument(virtualDocument: VirtualDocument): void;`
  - **After:** `unregisterDocument(uri: string): void;`

### CODE REVIEW

File: packages/lsp/src/connection_manager.ts [hunk 1]
```diff
@@ -653,7 +652,7 @@ namespace Private {
     capabilities: ClientCapabilities
   ): Promise<LSPConnection> {
     let connection = _connections.get(languageServerId);
-    if (connection == null) {
+    if (connection === undefined) {
```
Comment (trungleduc) [inline packages/lsp/src/connection_manager.ts:655]:
```suggestion
    if (connection) {
```

Comment (afshin) [reply packages/lsp/src/connection_manager.ts]: I changed it to `if (!connection)` since that's the condition I was replacing.

Comment (trungleduc) [reply packages/lsp/src/connection_manager.ts]: Ah sorry i misread the condition.

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @afshin!

**Comment by krassowski** [comment]
Kicked the visual regression tests - completer doc panel test was failing, let's see if that's a flake.

**Review by krassowski — APPROVED** [review]
Thank you!

**Pull request merged**